src/parser/mod.rs

```diff
@@ -1288,6 +1300,35 @@ impl<'a> Parser<'a> {
             _ => self.expected("an expression", next_token),
         }?;
 
+        // Composite function chain
+        while matches!(
+            expr,
+            Expr::Function(_)
+                | Expr::CompositeFunction { .. }
+                | Expr::Cast { .. }
+                | Expr::Convert { .. }
```

Comment:
hmm some of the entries seem a bit arbitary (like cast, convert)? I wonder if instead of having this block as a special case, piggy-backing on my other comment to do similar to the lambda parsing, we could probably have the chain continuation logic in the same helper function that gets called once here (as in the function knows to parse not just a single function but also continuation/suffix if present of the full expression)